scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -3137,22 +3143,42 @@ def _pass_thru(
     return res[-len(parent_spec):]
 
 
-def _raw_unit_addr(node: dtlib_Node) -> bytes:
+def _raw_unit_addr(node: dtlib_Node, parent: dtlib_Node) -> bytes:
     # _map_interrupt() helper. Returns the unit address (derived from 'reg' and
     # #address-cells) as a raw 'bytes'
 
+    iparent: Optional[dtlib_Node] = parent
+    iparent_addr_len = _address_cells_self(iparent)
+    node_addr_len = _address_cells_self(node)
```

Comment:
We need parent node address cells here.
```suggestion
    node_addr_len = _address_cells(node)
```